src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -813,24 +814,34 @@ void GLHEBase::simulate(EnergyPlusData &state,
 
 //******************************************************************************
 
-PlantComponent *GLHEBase::factory(EnergyPlusData &state, DataPlant::PlantEquipmentType objectType, std::string const &objectName)
+GLHEBase *GLHEBase::factory(EnergyPlusData &state, DataPlant::PlantEquipmentType objectType, std::string const &objectName)
 {
     if (state.dataGroundHeatExchanger->GetInput) {
         GetGroundHeatExchangerInput(state);
         state.dataGroundHeatExchanger->GetInput = false;
     }
     if (objectType == DataPlant::PlantEquipmentType::GrndHtExchgSystem) {
-        for (auto &ghx : state.dataGroundHeatExchanger->verticalGLHE) {
-            if (ghx.name == objectName) {
-                return &ghx;
-            }
-        }
+        auto thisObj = std::find_if(state.dataGroundHeatExchanger->verticalGLHE.begin(),
+                                    state.dataGroundHeatExchanger->verticalGLHE.end(),
+                                    [&objectName](const GLHEBase &myObj) { return myObj.name == objectName; });
+        if (thisObj != state.dataGroundHeatExchanger->verticalGLHE.end()) return &(*thisObj);
+
+        // for (auto &ghx : state.dataGroundHeatExchanger->verticalGLHE) {
+        //    if (ghx.name == objectName) {
+        //        return &ghx;
+        //    }
+        //}
     } else if (objectType == DataPlant::PlantEquipmentType::GrndHtExchgSlinky) {
-        for (auto &ghx : state.dataGroundHeatExchanger->slinkyGLHE) {
-            if (ghx.name == objectName) {
-                return &ghx;
-            }
-        }
```

Comment:
[src/EnergyPlus/GroundHeatExchangers.cc:830]:(style),[useStlAlgorithm],Consider using std::find_if algorithm instead of a raw loop.